tests/e2e/system_test/test_full_cluster_health.py

```diff
@@ -0,0 +1,286 @@
+"""
+Test to verify cluster health/stability when it's full (75%)
+"""
+import logging
+import time
+from ocs_ci.ocs.cluster import CephCluster, get_percent_used_capacity
+from ocs_ci.ocs import constants, benchmark_operator, ocp, defaults
+from ocs_ci.ocs.perftests import PASTest
+from ocs_ci.helpers.performance_lib import run_command
+from ocs_ci.utility import templating
+from ocs_ci.helpers.helpers import get_full_test_logs_path
+from ocs_ci.ocs.resources import pod
+from ocs_ci.ocs.disruptive_operations import osd_node_reboot
+from ocs_ci.ocs.node import wait_for_nodes_status
+from ocs_ci.framework.pytest_customization.marks import tier3
+
+logger = logging.getLogger(__name__)
+
+
+class TestFullClusterHealth(PASTest):
+    """
+    Test Cluster health when storage is ~75%
+    """
+
+    def setup(self):
+        """
+        Setting up test parameters
+        """
+        logger.info("Starting the test setup")
+        self.percent_to_fill = 75.0
+        self.ceph_cluster = CephCluster()
+        self.nodes = None
+
+        self.benchmark_name = "FIO"
+        self.client_pod_name = "fio-client"
+
+        super(TestFullClusterHealth, self).setup()
+        # deploy the benchmark-operator
+        self.deploy_benchmark_operator()
+
+    def teardown(self):
+        """
+        The teardown of the test environment in the end.
+
+        """
+
+        self.operator.cleanup()
+        logger.info("cleanup the environment")
+        sleep_time = 5
+        time.sleep(sleep_time * 60)
+        logger.info(
+            f"Going to sleep for {sleep_time} Minute, for background cleanup to complete"
+        )
+
+    def run(self):
+        """
+
+        Run the test, and wait until it finished
+        """
+
+        self.deploy_and_wait_for_wl_to_start(timeout=900)
+        self.wait_for_wl_to_finish(sleep=300)
+
+        try:
+            if "Fio failed to execute" not in self.test_logs:
+                logger.info("FIO has completed successfully")
+        except IOError:
+            logger.warning("FIO failed to complete")
+
+    def calculate_crd_data(self):
+        """
+        Getting the storage capacity and calculate pod count and pvc size
+
+        """
+
+        ceph_used_capacity_percent = get_percent_used_capacity()
+        logging.info(f"Ceph used capacity percent is {ceph_used_capacity_percent}%")
+
+        ceph_capacity = self.ceph_cluster.get_ceph_capacity()
+        logging.info(f"Total storage capacity is {ceph_capacity} GiB")
+
+        self.percent_to_fill = self.percent_to_fill - ceph_used_capacity_percent
```

Comment:
this sets a fixed amount of data, in percentage, to fill the cluster with. What if the cluster is getting filled by other applications, such as background IO? Then the cluster will be filled with more than the desired percentage. 
Would it be possible to watch the cluster used capacity while it is being filled instead of filling it with a fixed amount of data?

Reply:
I'm calculating the required size and number of servers at the time of execution.
In most cases, it works as expected.
We can divide it into smaller size running calculations before/after each writes.
IMO, we should do it in separate PR in order to prevent blocking this PR